src/custom/hooks/useErrorMessageAndModal.tsx

```diff
@@ -23,3 +23,30 @@ export default function useErrorMessage() {
       error ? <SwapCallbackError showClose={showClose} handleClose={handleCloseError} error={error} {...rest} /> : null,
   }
 }
+
+export function useErrorModal() {
+  // Any async bc errors
+  const [internalError, setError] = useState<string | undefined>()
+  const { openModal, closeModal, TransactionErrorModal } = useTransactionErrorModal()
+
+  const handleCloseError = () => {
+    closeModal()
+    setError(undefined)
+  }
+  const handleSetError = (error: string | undefined) => {
+    // close any open modals
+    // closeModal()
+    // set the error and IF error, open modal
+    setError(error)
+    error && openModal()
+  }
+
+  return {
+    error: internalError,
+    handleCloseError,
+    handleSetError,
+    ErrorModal: ({ message = internalError }: { message?: string }) => (
+      <TransactionErrorModal onDismiss={handleCloseError} message={message} />
+    ),
```

Comment:
should we have a `useCallback` for this component too like `TransactionErrorModal`?

Reply:
ah, yes good catch